plugins/connections/ui/src/main/java/org/pentaho/di/vfs/connections/ui/dialog/ConnectionDialog.java

```diff
@@ -299,6 +313,54 @@ private void updateConnectionType( String connectionType ) {
     wConnectionTypeComp.layout();
   }
 
+  private void addRootPathComponent() {
+    VFSConnectionDetails vfsConnectionDetails = asVFSConnectionDetails( connectionDetails );
+    if ( vfsConnectionDetails != null && vfsConnectionDetails.isSupportsRootPath() ) {
+      Control control = getLastWidgetFromParentComposite( wDetailsWrapperComp );
+      Label wlRootPath =  helper.createLabel( wDetailsWrapperComp, SWT.LEFT | SWT.WRAP, "ConnectionDialog.RootFolderPath.Label", control );
+      wRootPath = helper.createTextVar( variableSpace, wDetailsWrapperComp, TEXT_VAR_FLAGS, wlRootPath, 0 );
+      wRootPath.setText( Const.NVL( vfsConnectionDetails.getRootPath(), "" ) );
+      wRootPath.addModifyListener( modifyEvent -> vfsConnectionDetails.setRootPath( wRootPath.getText() ) );
+
+      if ( vfsDetailsComposite.canBrowseRootPath() ) {
+        addBrowseButtonForRootPathComponent();
+      }
+    }
+  }
+
+  private void addBrowseButtonForRootPathComponent() {
+    Button wBrowseButton = new Button( wDetailsWrapperComp, SWT.PUSH );
+    wBrowseButton.setText( BaseMessages.getString( PKG, "ConnectionDialog.Browse.Label" ) );
+    props.setLook( wBrowseButton );
+    wBrowseButton.setLayoutData( helper.getFormDataLabel( wRootPath ) );
+
+    Object object = vfsDetailsComposite.getRootPathSelectionAdapter( wRootPath );
+    if ( Objects.nonNull( object ) && ( object instanceof SelectionAdapterFileDialogTextVar ) ) {
```

Comment:
The test `Objects.nonNull( object ) &&` can be removed. `instanceof` already handles null values.